### ISSUE TITLE
Tests fail to compile with glibc >= 2.34

### ISSUE THREAD

**Issue by susnux** [fuzzylite/fuzzylite#94]
The bundled version of Catch is affected by this issue (`SIGSTKSZ` is not a constant anymore):  https://github.com/catchorg/Catch2/issues/2178

So tests do not compile with glibc > 2.34.

You could replace Catch with Catch2 (and require C++11 for tests), that will solve the issue.
I use this patch as a workaround (and use the system provided Catch2 version): [fix-tests.patch](https://github.com/fuzzylite/fuzzylite/files/7972235/fix-tests.patch.txt)

Edit:
If a patch header / authorship information required, please feel free to use this one:
```
From: Ferdinand Thiessen <rpm@fthiessen.de>
Date: Mon, 31 Jan 2022 16:06:19 GMT
Subject: Replace Catch with Catch2 and require C++11 for tests
```

**Comment by josch** [comment]
Thank you susnux for this patch! fuzzylite in Debian also suffers from the `SIGSTKSZ` issue with catch: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1017155

Could you post your patch with authorship information? I'd like to carry it in the Debian package and need to correctly attribute it for that. Thanks!

**Comment by josch** [comment]
Hi @susnux, I now used this as the patch header:

```
From: Ferdinand Thiessen <f.thiessen@gmx.de>
Date: Mon, 12 Sep 2022 07:18:39 GMT
Subject: Replace Catch with Catch2 and require C++11 for tests
```

Please chime in if that is not correct. Thanks!

**Comment by susnux** [comment]
@josch Feel free to use the header (edited the first comment), but please remove that other email address.

**Comment by josch** [comment]
> @josch Feel free to use the header (edited the first comment), but please remove that other email address.

Thank you! If you don't want that email address to be public, maybe you'd like to know that this address is publicly available in your other github repos.

**Comment by jcrada** [comment]
Thanks for patch and information.
Is it possible to just update the catch2 library? Instead of trying to find it via Cmake?

**Comment by josch** [comment]
If you do decide to bundle the catch2 library, know that in Debian we'll patch it out so that fuzzylite doesn't ship it. While it might be convenient for you, for distributions, every bundled library causes much work because every time a security issue is discovered, we have to then touch all the packages that happen to bundle the versions of the library affected by the security issue instead of just patching the library in question and calling it a day. This is also problematic because it's not always easy to find out which version of a library is being bundled. So the safe thing to do is to just avoid any bundling in the first place.

**Comment by jcrada** [comment]
I understand, @josch. However, as I apply the patch, I find myself searching the internet how to make Catch2 available via CMake to the project and there are a few options. 

What would be the easiest option that requires the least effort to fuzzylite users to install Catch2 non-globally? 

What about shipping catch2 as we are now, but using it only if `find_package(Catch2)` fails?

**Comment by josch** [comment]
> What would be the easiest option that requires the least effort to fuzzylite users to install Catch2 non-globally?

I guess the same way as installing any other cmake project non-globally by setting a custom installation prefix. Looking at how Debian compiles https://github.com/catchorg/Catch2 there seems to be no special magic involved to get this cmake project configured, built and installed.

> What about shipping catch2 as we are now, but using it only if find_package(Catch2) fails?

You can do that, no problem.

**Comment by jcrada** [comment]
Fixed in `development` branch: a651f3d13e192352d7a02ee9da65ab591a963ab9
